### ISSUE TITLE
yup npm package throws "Invalid Version: undefined" error

### ISSUE THREAD

**Issue by EvHaus** [flow-typed/flow-typed#3834]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

When you have the `yup` package as a dependency, ie:

```json
"dependencies": {
    "yup": "0.28.5"
}
```

And then you run `flow-typed install --ignoreDeps dev`, flow-typed crashes with:

```
• Found 1 dependencies in package.json to install libdefs for. Searching...
• rebasing flow-typed cache...
UNCAUGHT ERROR: TypeError: Invalid Version: undefined
    at new SemVer (/Users/evhaus/Git/myapp/node_modules/flow-typed/node_modules/semver/classes/semver.js:22:13)
    at compare (/Users/evhaus/Git/myapp/node_modules/flow-typed/node_modules/semver/functions/compare.js:3:3)
    at Object.lt (/Users/evhaus/Git/myapp/node_modules/flow-typed/node_modules/semver/functions/lt.js:2:29)
    at Promise.all.libDefsToSearchForEntries.map (/Users/evhaus/Git/myapp/node_modules/flow-typed/dist/commands/install.js:287:28)
error Command failed with exit code 1.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Reproducible with a tiny `package.json`:

```json
{
	"name": "test",
	"dependencies": {
		"yup": "0.28.5"
	},
	"devDependencies": {
		"flow-bin": "0.124.0",
		"flow-typed": "3.1.0"
	}
}
```

Then install via `npm install` and then run `npx flow-typed install --ignoreDeps dev`.

**What is the expected behavior?**

Libdefs should be installed, or ignored. But not a hard crash.

**Local Environment Information**

```
  System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
    Memory: 295.04 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v10.17.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    Homebrew: 2.2.16 - /usr/local/bin/brew
    pip3: 20.0.2 - /usr/local/bin/pip3
    RubyGems: 3.0.3 - /usr/bin/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 4.2.1 - /usr/bin/gcc
    Git: 2.26.2 - /usr/local/bin/git
    Clang: 1103.0.32.59 - /usr/bin/clang
    Subversion: 1.10.4 - /usr/bin/svn
  Servers:
    Apache: 2.4.41 - /usr/sbin/apachectl
  Virtualization:
    Docker: 19.03.8 - /usr/local/bin/docker
  SDKs:
    iOS SDK:
      Platforms: iOS 13.4, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
  IDEs:
    Atom: 1.47.0-beta0
    Nano: 2.0.6 - /usr/bin/nano
    VSCode: 1.44.2 - /usr/local/bin/code
    Vim: 8.1 - /usr/bin/vim
    Xcode: 11.4.1/11E503a - /usr/bin/xcodebuild
  Languages:
    Bash: 3.2.57 - /bin/bash
    Go: 1.14.2 - /usr/local/bin/go
    Perl: 5.18.4 - /usr/bin/perl
    PHP: 7.3.11 - /usr/bin/php
    Python: 2.7.16 - /usr/bin/python
    Python3: 3.7.7 - /usr/local/bin/python3
    Ruby: 2.6.3 - /usr/bin/ruby
  Databases:
    PostgreSQL: 12.2 - /usr/local/bin/postgres
    SQLite: 3.28.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 81.0.4044.138
    Chrome Canary: 84.0.4144.2
    Firefox: 76.0
    Safari: 13.1
```

**Comment by osmestad** [comment]
I'm seeing the same error, I have not isolated which dependencies it fails on here, but the console output I get is:

```
$ yarn flow-typed update --skip
$ /Users/osmestad/Code/wimp-web-client/node_modules/.bin/flow-typed update --skip
• Found 4 dependencies in package.json to install libdefs for. Searching...
• rebasing flow-typed cache...
UNCAUGHT ERROR: TypeError: Invalid Version: undefined
    at new SemVer (/Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/node_modules/semver/classes/semver.js:22:13)
    at compare (/Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/node_modules/semver/functions/compare.js:3:3)
    at Object.lt (/Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/node_modules/semver/functions/lt.js:2:29)
    at /Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/dist/commands/install.js:287:28
    at async Promise.all (index 2)
    at async installNpmLibDefs (/Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/dist/commands/install.js:268:3)
    at async Object.run (/Users/osmestad/Code/wimp-web-client/node_modules/flow-typed/dist/commands/install.js:149:27)
```

**Comment by sampi** [comment]
I'm getting the same error, but I am having a hard time figuring out which package causes the issue. some extra debug logging would help a lot.

`flow-bin@0.125.1`
`flow-typed@3.1.0`

```
​> flow-typed update --verbose
• Found package.json dependency: cross-env@^7.0.0
• Found package.json dependency: @adeira/babel-preset-adeira@^0.4.0
• Found package.json dependency: @adeira/eslint-config@^2.0.0
• Found package.json dependency: @adeira/eslint-runner@^0.1.0
• Found package.json dependency: @adeira/flow-bin@^0.2.0
• Found package.json dependency: @babel/cli@^7.7.5
• Found package.json dependency: @babel/core@^7.7.5
• Found package.json dependency: @babel/node@^7.7.4
• Found package.json dependency: @babel/plugin-proposal-class-properties@^7.7.4
• Found package.json dependency: @babel/plugin-proposal-nullish-coalescing-operator@^7.7.4
• Found package.json dependency: @babel/plugin-proposal-optional-chaining@^7.7.5
• Found package.json dependency: @babel/plugin-transform-modules-commonjs@^7.7.5
• Found package.json dependency: @babel/plugin-transform-runtime@^7.7.6
• Found package.json dependency: @babel/preset-env@^7.7.6
• Found package.json dependency: @babel/preset-flow@^7.7.4
• Found package.json dependency: @babel/preset-react@^7.7.4
• Found package.json dependency: @knapsack-pro/cypress@^2.0.0
• Found package.json dependency: @kiwicom/translations@^1.601.87
• Found package.json dependency: @testing-library/react@^10.0.0
• Found package.json dependency: babel-eslint@^10.0.1
• Found package.json dependency: babel-jest@^26.0.0
• Found package.json dependency: babel-loader@^8.0.6
• Found package.json dependency: babel-plugin-id@^1.0.1
• Found package.json dependency: babel-plugin-relay@~9.1.0
• Found package.json dependency: babel-plugin-styled-components@^1.10.7
• Found package.json dependency: babel-plugin-transform-assets-import-to-string@^1.2.0
• Found package.json dependency: babel-plugin-wildcard@^6.0.0
• Found package.json dependency: concurrently@^5.1.0
• Found package.json dependency: core-js@^3.4.8
• Found package.json dependency: cypress@4.7.0
• Found package.json dependency: cypress-terminal-report@^1.2.1
• Found package.json dependency: date-fns@^2.11.1
• Found package.json dependency: dotenv@^8.0.0
• Found package.json dependency: edit-json-file@^1.3.0
• Found package.json dependency: eslint@^7.0.0
• Found package.json dependency: eslint-plugin-cypress@^2.8.1
• Found package.json dependency: eslint-plugin-relay-fragments@^0.0.2
• Found package.json dependency: flow-bin@^0.125.0
• Found package.json dependency: flow-copy-source@^2.0.0
• Found package.json dependency: flow-typed@^3.0.0
• Found package.json dependency: graphql-cli@^3.0.14
• Found package.json dependency: jest@^26.0.0
• Found package.json dependency: jest-environment-jsdom@^26.0.0
• Found package.json dependency: jest-environment-jsdom-global@^2.0.0
• Found package.json dependency: jest-junit@^10.0.0
• Found package.json dependency: prettier-eslint@^10.0.0
• Found package.json dependency: relay-compiler@~9.1.0
• Found package.json dependency: wsrun@^5.2.0
UNCAUGHT ERROR: TypeError: Invalid Version: undefined
    at new SemVer (/my/package/path/node_modules/flow-typed/node_modules/semver/classes/semver.js:22:13)
    at outside (/my/package/path/node_modules/flow-typed/node_modules/semver/ranges/outside.js:12:13)
    at Object.gtr (/my/package/path/node_modules/flow-typed/node_modules/semver/ranges/gtr.js:3:42)
    at pkgVersionMatch (/my/package/path/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:260:42)
    at /my/package/path/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:279:82
    at Array.filter (<anonymous>)
    at filterLibDefs (/my/package/path/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:273:15)
    at findNpmLibDef (/my/package/path/node_modules/flow-typed/dist/lib/npm/npmLibDefs.js:327:27)
    at async /my/package/path/node_modules/flow-typed/dist/commands/install.js:273:20
    at async Promise.all (index 37)
    at async installNpmLibDefs (/my/package/path/node_modules/flow-typed/dist/commands/install.js:268:3)
    at async Object.run (/my/package/path/node_modules/flow-typed/dist/commands/install.js:149:27)
```

**Comment by osmestad** [comment]
Reverting `flow-typed` to: `2.6.2` seems to make it run again, so I assume it is a regression introduced in `3.0.0` (which fails like `3.1.0` for me).

**Comment by bryan-lifelink** [comment]
I was able to get it to work by hacking the npmLibDefs and dist (around the line numbers indicated in the error message) in the flow-typed directory to default to 0.0.0 for the package version when semver returned undefined for the lower bound.  It looks like it chokes when there isn't a major version less than the current, which seems to happen with package versions less than 1.0.0.

**Comment by pascalduez** [comment]
Looks similar to #3809

Could some of you try with `flow-typed@latest` and report whether the problem persist?

**Comment by EvHaus** [comment]
Confirmed. My issue is now resolved (using `flow-typed@3.2.1`).
